### PR TITLE
Bundle with hash devas 1946

### PR DESCRIPTION
**Pull request by amensouissi**

### CODE REVIEW

File: assembl/views/__init__.py [hunk 1]
```diff
@@ -76,8 +84,10 @@ def find_theme(theme_name, frontend_version=1):
     @returns the theme path fragment relative to the theme base_path, or
     None if not found
     """
-    theme_base_path = get_theme_base_path(frontend_version)
+    current_theme = THEMES.get(theme_name, None)
+    if current_theme: return current_theme
```
Comment (vincentfretin) [inline assembl/views/__init__.py]: I don't like one liner like this, please add a new line and indent.

File: assembl/views/__init__.py [hunk 2]
```diff
@@ -86,7 +96,8 @@ def find_theme(theme_name, frontend_version=1):
             (head, name) = os.path.split(dirpath)
             print name, relpath
             if name == theme_name:
-                return relpath
+                THEMES[theme_name] = relpath
+                return THEMES[theme_name]
```
Comment (vincentfretin) [inline assembl/views/__init__.py]: You need two caches based on frontend_version. THEMES[frontend_version][theme_name]

Comment (vincentfretin) [reply assembl/views/__init__.py]: You can `return relpath` instead of accessing the dict.

File: assembl/views/__init__.py [hunk 3]
```diff
@@ -110,6 +121,45 @@ def get_theme_info(discussion, frontend_version=1):
         return ('default', 'default')
 
 
+def get_resources_hash(theme_name):
+    current_resources = RESOURCES.get(theme_name, None)
+    if current_resources: return current_resources
```
Comment (vincentfretin) [inline assembl/views/__init__.py]: please add a newline

File: assembl/views/__init__.py [hunk 4]
```diff
@@ -110,6 +121,45 @@ def get_theme_info(discussion, frontend_version=1):
...
+
+    resources_base_path = os.path.join(
+        os.path.dirname(os.path.dirname(__file__)),
+        'static2', 'build', 'index.html')
```
Comment (vincentfretin) [inline assembl/views/__init__.py]: You should rename the variable `resources_base_path` to `resources_path`

File: assembl/views/__init__.py [hunk 5]
```diff
@@ -110,6 +121,45 @@ def get_theme_info(discussion, frontend_version=1):
...
+            soup = BeautifulSoup(fp)
+            bundle = soup.find(src=get_bundle_hash)
+            bundle_css = soup.find(href=get_bundle_css_hash)
+            theme_css = soup.find(href=get_theme_css_hash)
+            result['bundle_hash'] = get_bundle_hash(bundle['src']).group(1) if bundle else None
+            result['bundle_css_hash'] = get_bundle_css_hash(bundle_css['href']).group(1) if bundle_css else None
+            result['theme_hash'] = get_theme_css_hash(theme_css['href']).group(1) if theme_css else None
+    
+    RESOURCES[theme_name] = result
+    return RESOURCES[theme_name]
```
Comment (vincentfretin) [inline assembl/views/__init__.py]: `return result`

File: assembl/views/discussion/views.py [hunk 6]
```diff
@@ -233,6 +234,8 @@ def react_view(request, required_permission=P_READ):
         "sentry_dsn": get('sentry_dsn', ''),
         "bugherd_url": bugherd_url
     }
+    resources_hash = get_resources_hash(theme_name)
+    common_context.update(resources_hash)
```
Comment (vincentfretin) [inline assembl/views/discussion/views.py]: You should put this in `get_default_context`, not here. You need to handle login view too.

File: assembl/views/__init__.py [hunk 7]
```diff
@@ -110,6 +121,45 @@ def get_theme_info(discussion, frontend_version=1):
         return ('default', 'default')
 
 
+def get_resources_hash(theme_name):
```
Comment (cedricmessiant) [inline assembl/views/__init__.py]: you could write a test for this function ;-)

File: assembl/views/__init__.py (hunk 1 above)
Comment (cedricmessiant) [reply assembl/views/__init__.py]: :+1:

File: assembl/views/__init__.py [hunk 8]
```diff
@@ -110,6 +121,45 @@ def get_theme_info(discussion, frontend_version=1):
...
+        'bundle_css_hash': None,
+        'theme_hash': None
+    }
+    if os.path.exists(resources_base_path):
+        with open(resources_base_path) as fp:
+            soup = BeautifulSoup(fp)
+            bundle = soup.find(src=get_bundle_hash)
+            bundle_css = soup.find(href=get_bundle_css_hash)
+            theme_css = soup.find(href=get_theme_css_hash)
+            result['bundle_hash'] = get_bundle_hash(bundle['src']).group(1) if bundle else None
```
[the remark this answers is not in the document]
Comment (amensouissi) [reply assembl/views/__init__.py]: `bundle = soup.find(src=get_bundle_hash)` ==> src in bundle if bundle

[the remark this answers is not in the document]
Comment (ayyazdaniaryan) [reply assembl/views/__init__.py]: Ahh, then carry on :)

**Pull request merged**